src/nplinker/metabolomics/molecular_family.py

```diff
@@ -15,21 +25,31 @@ def __init__(self, family_id):
     #     return False
 
     @property
-    def strains(self):
-        strains = set()
+    def strains(self) -> set[StrainCollection]:
+        """Get strains of spectra in the molecular family.
+
+        Returns:
+            set[StrainCollection]: StrainCollection of strains from which the spectra in the molecular family are coming.
+        """
+        strains: set[StrainCollection] = set()
         for spectrum in self.spectra:
             strains = strains.union(spectrum.strains)
```

Comment:
```suggestion
            strains.add(spectrum.strains)
```
Note that `spectrum.strains` is not a set but an element, so `set.union` cannot be used

Reply:
Interestingly, the results of nplinker change with this commit - any idea why?

Reply:
The former `strains = strains.union(spectrum.strains)` is actually a set of `Strain` objects but not `StrainCollection` objects. 
The current `strains.add(spectrum.strains)` will generate a set of `StrainCollection` objects. 
If there is no further `union` operation on this `strains`, redundant `Strain` objects from different `StrainCollection` may give problems.

You need to check where this `strains` is used to figure out which type of objects is actually needed.